### ISSUE TITLE
Running eslint hangs

### ISSUE THREAD

**Issue by sebdanielsson** [oven-sh/bun#11801]
### What version of Bun is running?

1.1.13+bd6a60512

### What platform is your computer?

Darwin 23.5.0 arm64 arm

### What steps can reproduce the bug?

packages:

```json
"devDependencies": {
    "@typescript-eslint/parser": "^7.13.0",
    "eslint": "^8.57.0",
    "eslint-plugin-astro": "^1.2.2",
    "postcss-html": "^1.7.0",
    "prettier": "^3.3.2",
    "prettier-plugin-astro": "^0.14.0",
    "prettier-plugin-tailwindcss": "^0.6.3",
    "stylelint": "^16.6.1",
    "stylelint-config-html": "^1.1.0",
    "stylelint-config-standard": "^36.0.0",
    "typescript": "^5.4.5",
    "typescript-eslint": "^7.13.0"
  }
```

.eslintrc.cjs:

```js
/** @type {import("@types/eslint").Linter.Config} */
module.exports = {
  extends: ['plugin:astro/recommended', 'plugin:@typescript-eslint/strict'],
  overrides: [
    {
      files: ['*.astro'],
      parser: 'astro-eslint-parser',
      parserOptions: {
        parser: '@typescript-eslint/parser',
        extraFileExtensions: ['.astro'],
      },
      rules: {},
    },
  ],
};
```

Command:

```shell
bunx --bun eslint "src/**/*.{astro,mjs,cjs,js,ts}"
```

### What is the expected behavior?

Linting should run.

### What do you see instead?

Hangs at line:

```
2024-06-12T08:57:16.484Z eslint:linter Apply the processor: 'astro/.astro'
```

Command: `DEBUG=eslint:* bunx --bun eslint "src/**/*.{astro,mjs,cjs,js,ts}"`

<details>
  <summary>Show logs</summary>
  
```
2024-06-12T08:57:16.088Z eslint:cli CLI args: [ 'src/**/*.{astro,mjs,cjs,js,ts}' ]
2024-06-12T08:57:16.096Z eslint:cli Using flat config? false
2024-06-12T08:57:16.099Z eslint:cli Running on files
2024-06-12T08:57:16.108Z eslint:file-enumerator Start to iterate files: [ 'src/**/*.{astro,mjs,cjs,js,ts}' ]
2024-06-12T08:57:16.108Z eslint:file-enumerator Glob: src/**/*.{astro,mjs,cjs,js,ts}
2024-06-12T08:57:16.109Z eslint:file-enumerator recursive? true
2024-06-12T08:57:16.109Z eslint:file-enumerator Enter the directory: /Users/sebastian/Git/sebdanielsson.github.io/src
2024-06-12T08:57:16.361Z eslint:rules Loading rule 'block-spacing' (remaining=290)
2024-06-12T08:57:16.364Z eslint:rules Loading rule 'brace-style' (remaining=289)
2024-06-12T08:57:16.365Z eslint:rules Loading rule 'comma-dangle' (remaining=288)
2024-06-12T08:57:16.367Z eslint:rules Loading rule 'consistent-return' (remaining=287)
2024-06-12T08:57:16.372Z eslint:rules Loading rule 'dot-notation' (remaining=286)
2024-06-12T08:57:16.374Z eslint:rules Loading rule 'indent' (remaining=285)
2024-06-12T08:57:16.375Z eslint:rules Loading rule 'init-declarations' (remaining=284)
2024-06-12T08:57:16.376Z eslint:rules Loading rule 'key-spacing' (remaining=283)
2024-06-12T08:57:16.377Z eslint:rules Loading rule 'keyword-spacing' (remaining=282)
2024-06-12T08:57:16.378Z eslint:rules Loading rule 'lines-around-comment' (remaining=281)
2024-06-12T08:57:16.379Z eslint:rules Loading rule 'lines-between-class-members' (remaining=280)
2024-06-12T08:57:16.380Z eslint:rules Loading rule 'max-params' (remaining=279)
2024-06-12T08:57:16.387Z eslint:rules Loading rule 'no-dupe-class-members' (remaining=278)
2024-06-12T08:57:16.388Z eslint:rules Loading rule 'no-empty-function' (remaining=277)
2024-06-12T08:57:16.390Z eslint:rules Loading rule 'no-extra-parens' (remaining=276)
2024-06-12T08:57:16.391Z eslint:rules Loading rule 'no-extra-semi' (remaining=275)
2024-06-12T08:57:16.395Z eslint:rules Loading rule 'no-invalid-this' (remaining=274)
2024-06-12T08:57:16.396Z eslint:rules Loading rule 'no-loop-func' (remaining=273)
2024-06-12T08:57:16.396Z eslint:rules Loading rule 'no-loss-of-precision' (remaining=272)
2024-06-12T08:57:16.396Z eslint:rules Loading rule 'no-magic-numbers' (remaining=271)
2024-06-12T08:57:16.404Z eslint:rules Loading rule 'no-restricted-imports' (remaining=270)
2024-06-12T08:57:16.419Z eslint:rules Loading rule 'no-unused-expressions' (remaining=269)
2024-06-12T08:57:16.420Z eslint:rules Loading rule 'no-useless-constructor' (remaining=268)
2024-06-12T08:57:16.422Z eslint:rules Loading rule 'object-curly-spacing' (remaining=267)
2024-06-12T08:57:16.425Z eslint:rules Loading rule 'prefer-destructuring' (remaining=266)
2024-06-12T08:57:16.441Z eslint:rules Loading rule 'quotes' (remaining=265)
2024-06-12T08:57:16.444Z eslint:rules Loading rule 'semi' (remaining=264)
2024-06-12T08:57:16.445Z eslint:rules Loading rule 'space-before-blocks' (remaining=263)
2024-06-12T08:57:16.446Z eslint:rules Loading rule 'space-infix-ops' (remaining=262)
2024-06-12T08:57:16.454Z eslint:rules Loading rule 'constructor-super' (remaining=261)
2024-06-12T08:57:16.455Z eslint:rules Loading rule 'getter-return' (remaining=260)
2024-06-12T08:57:16.455Z eslint:rules Loading rule 'no-const-assign' (remaining=259)
2024-06-12T08:57:16.455Z eslint:rules Loading rule 'no-dupe-args' (remaining=258)
2024-06-12T08:57:16.455Z eslint:rules Loading rule 'no-dupe-keys' (remaining=257)
2024-06-12T08:57:16.455Z eslint:rules Loading rule 'no-func-assign' (remaining=256)
2024-06-12T08:57:16.456Z eslint:rules Loading rule 'no-import-assign' (remaining=255)
2024-06-12T08:57:16.456Z eslint:rules Loading rule 'no-new-symbol' (remaining=254)
2024-06-12T08:57:16.456Z eslint:rules Loading rule 'no-new-native-nonconstructor' (remaining=253)
2024-06-12T08:57:16.456Z eslint:rules Loading rule 'no-obj-calls' (remaining=252)
2024-06-12T08:57:16.456Z eslint:rules Loading rule 'no-redeclare' (remaining=251)
2024-06-12T08:57:16.457Z eslint:rules Loading rule 'no-setter-return' (remaining=250)
2024-06-12T08:57:16.457Z eslint:rules Loading rule 'no-this-before-super' (remaining=249)
2024-06-12T08:57:16.457Z eslint:rules Loading rule 'no-undef' (remaining=248)
2024-06-12T08:57:16.457Z eslint:rules Loading rule 'no-unreachable' (remaining=247)
2024-06-12T08:57:16.457Z eslint:rules Loading rule 'no-unsafe-negation' (remaining=246)
2024-06-12T08:57:16.458Z eslint:rules Loading rule 'no-var' (remaining=245)
2024-06-12T08:57:16.458Z eslint:rules Loading rule 'prefer-const' (remaining=244)
2024-06-12T08:57:16.459Z eslint:rules Loading rule 'prefer-rest-params' (remaining=243)
2024-06-12T08:57:16.459Z eslint:rules Loading rule 'prefer-spread' (remaining=242)
2024-06-12T08:57:16.462Z eslint:rules Loading rule 'no-array-constructor' (remaining=241)
2024-06-12T08:57:16.464Z eslint:rules Loading rule 'no-unused-vars' (remaining=240)
2024-06-12T08:57:16.466Z eslint:file-enumerator Didn't match: .DS_Store
2024-06-12T08:57:16.467Z eslint:file-enumerator Yield: env.d.ts
2024-06-12T08:57:16.467Z eslint:cli-engine Lint /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.467Z eslint:linter Linting code for /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts (pass 1)
2024-06-12T08:57:16.468Z eslint:linter Verify
2024-06-12T08:57:16.468Z eslint:linter With ConfigArray: /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.468Z eslint:linter Parsing: /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.474Z eslint:linter Parsing successful: /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.475Z eslint:linter Scope analysis: /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.475Z eslint:linter Scope analysis successful: /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts
2024-06-12T08:57:16.481Z eslint:code-path onCodePathStart s1
2024-06-12T08:57:16.481Z eslint:code-path onCodePathSegmentStart s1_1
2024-06-12T08:57:16.481Z eslint:code-path s1_1) Program
2024-06-12T08:57:16.482Z eslint:code-path s1_1) Program:exit
2024-06-12T08:57:16.483Z eslint:code-path onCodePathSegmentEnd s1_1
2024-06-12T08:57:16.483Z eslint:code-path onCodePathEnd s1
2024-06-12T08:57:16.483Z eslint:code-path DOT 
digraph {
node[shape=box,style="rounded,filled",fillcolor=white];
initial[label="",shape=circle,style=filled,fillcolor=black,width=0.25,height=0.25];
final[label="",shape=doublecircle,style=filled,fillcolor=black,width=0.25,height=0.25];
s1_1[label="Program"];
initial->s1_1->final;
}
2024-06-12T08:57:16.483Z eslint:linter Generating fixed text for /Users/sebastian/Git/sebdanielsson.github.io/src/env.d.ts (pass 1)
2024-06-12T08:57:16.483Z eslint:source-code-fixer Applying fixes
2024-06-12T08:57:16.483Z eslint:source-code-fixer shouldFix parameter was false, not attempting fixes
2024-06-12T08:57:16.483Z eslint:file-enumerator Enter the directory: /Users/sebastian/Git/sebdanielsson.github.io/src/components
2024-06-12T08:57:16.484Z eslint:file-enumerator Yield: RaycastStoreCard.astro
2024-06-12T08:57:16.484Z eslint:cli-engine Lint /Users/sebastian/Git/sebdanielsson.github.io/src/components/RaycastStoreCard.astro
2024-06-12T08:57:16.484Z eslint:linter Linting code for /Users/sebastian/Git/sebdanielsson.github.io/src/components/RaycastStoreCard.astro (pass 1)
2024-06-12T08:57:16.484Z eslint:linter Verify
2024-06-12T08:57:16.484Z eslint:linter With ConfigArray: /Users/sebastian/Git/sebdanielsson.github.io/src/components/RaycastStoreCard.astro
2024-06-12T08:57:16.484Z eslint:linter Apply the processor: 'astro/.astro'
```
</details>

### Additional information

Works fine when using npx or bunx without `--bun`.

**Comment by rcerljenko** [comment]
same here

**Comment by vladdenisov** [comment]
***TLDR**: Problem is typically the same, just different config*

### What version of Bun is running?
1.1.13

### What platform is your computer?
Darwin 23.5.0 arm64 arm

Same problem appears on CI machine with amd64 cpu.

### What steps can reproduce the bug?

Dev dependencies:

```json
	"@typescript-eslint/eslint-plugin": "6.12.0",
    "@typescript-eslint/parser": "6.12.0",
	"eslint": "8.57.0",
    "eslint-config-prettier": "9.0.0",
    "eslint-formatter-gitlab": "5.1.0",
    "eslint-import-resolver-alias": "1.1.2",
    "eslint-import-resolver-typescript": "3.6.1",
    "eslint-plugin-filenames": "1.3.2",
    "eslint-plugin-import": "2.29.0",
    "eslint-plugin-jest-dom": "5.2.0",
    "eslint-plugin-prettier": "5.0.1",
    "eslint-plugin-react": "7.33.2",
    "eslint-plugin-react-hooks": "4.6.0",
    "eslint-plugin-sonarjs": "0.23.0",
    "eslint-plugin-storybook": "0.6.15",
    "prettier": "3.1.0",
  ```
  
  
  Eslint Config:
  
  ```js
  module.exports = {
	  env: {
	    browser: true,
	    es6: false,
	  },
	  extends: [
	    'eslint:recommended',
	    'plugin:@typescript-eslint/recommended',
	    'plugin:@typescript-eslint/eslint-recommended',
	    'plugin:@typescript-eslint/recommended-requiring-type-checking',
	    'plugin:prettier/recommended',
	    'plugin:import/recommended',
	    'plugin:import/typescript',
	    'plugin:sonarjs/recommended',
	    'plugin:react/recommended',
	    'plugin:storybook/recommended',
	    'plugin:react-hooks/recommended',
	  ],
	  plugins: ['@typescript-eslint', 'prettier', 'import', 'sonarjs'],
	  parser: '@typescript-eslint/parser',
	  parserOptions: {
	    sourceType: 'module',
	    project: './tsconfig.json',
	    tsconfigRootDir: '.',
	  },
	  settings: {
	    react: {
	      version: 'detect',
	    },
	    'import/resolver': {
	      typescript: {
	        project: './tsconfig.json',
	      },
	    },
	  },
	  ...project-specific rules
  }
```


Run the command:

`bunx --bun eslint src`




### What is the expected behavior?
Linting should run.

### What do you see instead?
When running using command `DEBUG=eslint:* bunx --bun eslint src`, hangs at line:

```logs
eslint:code-path onCodePathStart s1 +0ms
eslint:code-path onCodePathSegmentStart s1_1 +1ms
eslint:code-path s1_1) Program +1ms
```

<details>
<summary>Full log</summary>

```
DEBUG=eslint:* bunx --bun eslint src
  eslint:cli CLI args: [ 'src' ] +0ms
  eslint:cli Using flat config? false +12ms
  eslint:cli Running on files +3ms
  eslint:file-enumerator Start to iterate files: [ 'src' ] +0ms
  eslint:file-enumerator Directory: /Users/vladdenisov/projects/pspod/kgnc/ui/src +0ms
  eslint:file-enumerator Enter the directory: /Users/vladdenisov/projects/pspod/kgnc/ui/src +0ms
  eslint:rules Loading rule 'block-spacing' (remaining=290) +0ms
  eslint:rules Loading rule 'brace-style' (remaining=289) +6ms
  eslint:rules Loading rule 'comma-dangle' (remaining=288) +2ms
  eslint:rules Loading rule 'dot-notation' (remaining=287) +8ms
  eslint:rules Loading rule 'indent' (remaining=286) +4ms
  eslint:rules Loading rule 'init-declarations' (remaining=285) +41ms
  eslint:rules Loading rule 'key-spacing' (remaining=284) +5ms
  eslint:rules Loading rule 'keyword-spacing' (remaining=283) +11ms
  eslint:rules Loading rule 'lines-around-comment' (remaining=282) +6ms
  eslint:rules Loading rule 'lines-between-class-members' (remaining=281) +2ms
  eslint:rules Loading rule 'max-params' (remaining=280) +1ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=279) +22ms
  eslint:rules Loading rule 'no-empty-function' (remaining=278) +2ms
  eslint:rules Loading rule 'no-extra-parens' (remaining=277) +3ms
  eslint:rules Loading rule 'no-extra-semi' (remaining=276) +3ms
  eslint:rules Loading rule 'no-invalid-this' (remaining=275) +7ms
  eslint:rules Loading rule 'no-loop-func' (remaining=274) +2ms
  eslint:rules Loading rule 'no-loss-of-precision' (remaining=273) +1ms
  eslint:rules Loading rule 'no-magic-numbers' (remaining=272) +1ms
  eslint:rules Loading rule 'no-restricted-imports' (remaining=271) +11ms
  eslint:rules Loading rule 'no-unused-expressions' (remaining=270) +22ms
  eslint:rules Loading rule 'no-useless-constructor' (remaining=269) +3ms
  eslint:rules Loading rule 'object-curly-spacing' (remaining=268) +3ms
  eslint:rules Loading rule 'prefer-destructuring' (remaining=267) +3ms
  eslint:rules Loading rule 'quotes' (remaining=266) +37ms
  eslint:rules Loading rule 'semi' (remaining=265) +12ms
  eslint:rules Loading rule 'space-before-blocks' (remaining=264) +2ms
  eslint:rules Loading rule 'space-infix-ops' (remaining=263) +2ms
  eslint:rules Loading rule 'constructor-super' (remaining=262) +432ms
  eslint:rules Loading rule 'for-direction' (remaining=261) +2ms
  eslint:rules Loading rule 'getter-return' (remaining=260) +0ms
  eslint:rules Loading rule 'no-async-promise-executor' (remaining=259) +2ms
  eslint:rules Loading rule 'no-case-declarations' (remaining=258) +0ms
  eslint:rules Loading rule 'no-class-assign' (remaining=257) +0ms
  eslint:rules Loading rule 'no-compare-neg-zero' (remaining=256) +1ms
  eslint:rules Loading rule 'no-cond-assign' (remaining=255) +0ms
  eslint:rules Loading rule 'no-const-assign' (remaining=254) +1ms
  eslint:rules Loading rule 'no-constant-condition' (remaining=253) +0ms
  eslint:rules Loading rule 'no-control-regex' (remaining=252) +1ms
  eslint:rules Loading rule 'no-debugger' (remaining=251) +0ms
  eslint:rules Loading rule 'no-delete-var' (remaining=250) +0ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=249) +1ms
  eslint:rules Loading rule 'no-dupe-else-if' (remaining=248) +0ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=247) +0ms
  eslint:rules Loading rule 'no-duplicate-case' (remaining=246) +0ms
  eslint:rules Loading rule 'no-empty' (remaining=245) +0ms
  eslint:rules Loading rule 'no-empty-character-class' (remaining=244) +1ms
  eslint:rules Loading rule 'no-empty-pattern' (remaining=243) +0ms
  eslint:rules Loading rule 'no-ex-assign' (remaining=242) +1ms
  eslint:rules Loading rule 'no-extra-boolean-cast' (remaining=241) +0ms
  eslint:rules Loading rule 'no-fallthrough' (remaining=240) +1ms
  eslint:rules Loading rule 'no-func-assign' (remaining=239) +1ms
  eslint:rules Loading rule 'no-global-assign' (remaining=238) +1ms
  eslint:rules Loading rule 'no-import-assign' (remaining=237) +1ms
  eslint:rules Loading rule 'no-inner-declarations' (remaining=236) +0ms
  eslint:rules Loading rule 'no-invalid-regexp' (remaining=235) +0ms
  eslint:rules Loading rule 'no-irregular-whitespace' (remaining=234) +1ms
  eslint:rules Loading rule 'no-misleading-character-class' (remaining=233) +1ms
  eslint:rules Loading rule 'no-mixed-spaces-and-tabs' (remaining=232) +3ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=231) +1ms
  eslint:rules Loading rule 'no-nonoctal-decimal-escape' (remaining=230) +0ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=229) +1ms
  eslint:rules Loading rule 'no-octal' (remaining=228) +0ms
  eslint:rules Loading rule 'no-prototype-builtins' (remaining=227) +1ms
  eslint:rules Loading rule 'no-redeclare' (remaining=226) +0ms
  eslint:rules Loading rule 'no-regex-spaces' (remaining=225) +0ms
  eslint:rules Loading rule 'no-self-assign' (remaining=224) +1ms
  eslint:rules Loading rule 'no-setter-return' (remaining=223) +1ms
  eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=222) +0ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=221) +1ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=220) +0ms
  eslint:rules Loading rule 'no-undef' (remaining=219) +0ms
  eslint:rules Loading rule 'no-unexpected-multiline' (remaining=218) +1ms
  eslint:rules Loading rule 'no-unreachable' (remaining=217) +0ms
  eslint:rules Loading rule 'no-unsafe-finally' (remaining=216) +1ms
  eslint:rules Loading rule 'no-unsafe-negation' (remaining=215) +0ms
  eslint:rules Loading rule 'no-unsafe-optional-chaining' (remaining=214) +1ms
  eslint:rules Loading rule 'no-unused-labels' (remaining=213) +1ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=212) +0ms
  eslint:rules Loading rule 'no-useless-backreference' (remaining=211) +2ms
  eslint:rules Loading rule 'no-useless-catch' (remaining=210) +1ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=209) +0ms
  eslint:rules Loading rule 'no-with' (remaining=208) +1ms
  eslint:rules Loading rule 'require-yield' (remaining=207) +0ms
  eslint:rules Loading rule 'use-isnan' (remaining=206) +0ms
  eslint:rules Loading rule 'valid-typeof' (remaining=205) +1ms
  eslint:rules Loading rule 'no-var' (remaining=204) +1ms
  eslint:rules Loading rule 'prefer-const' (remaining=203) +1ms
  eslint:rules Loading rule 'prefer-rest-params' (remaining=202) +1ms
  eslint:rules Loading rule 'prefer-spread' (remaining=201) +0ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=200) +3ms
  eslint:rules Loading rule 'no-implied-eval' (remaining=199) +3ms
  eslint:rules Loading rule 'require-await' (remaining=198) +1ms
  eslint:rules Loading rule 'curly' (remaining=197) +2ms
  eslint:rules Loading rule 'max-len' (remaining=196) +1ms
  eslint:rules Loading rule 'no-confusing-arrow' (remaining=195) +1ms
  eslint:rules Loading rule 'no-mixed-operators' (remaining=194) +0ms
  eslint:rules Loading rule 'no-tabs' (remaining=193) +1ms
  eslint:rules Loading rule 'array-bracket-newline' (remaining=192) +0ms
  eslint:rules Loading rule 'array-bracket-spacing' (remaining=191) +1ms
  eslint:rules Loading rule 'array-element-newline' (remaining=190) +0ms
  eslint:rules Loading rule 'arrow-parens' (remaining=189) +1ms
  eslint:rules Loading rule 'arrow-spacing' (remaining=188) +0ms
  eslint:rules Loading rule 'comma-spacing' (remaining=187) +0ms
  eslint:rules Loading rule 'comma-style' (remaining=186) +1ms
  eslint:rules Loading rule 'computed-property-spacing' (remaining=185) +0ms
  eslint:rules Loading rule 'dot-location' (remaining=184) +1ms
  eslint:rules Loading rule 'eol-last' (remaining=183) +0ms
  eslint:rules Loading rule 'func-call-spacing' (remaining=182) +0ms
  eslint:rules Loading rule 'function-call-argument-newline' (remaining=181) +1ms
  eslint:rules Loading rule 'function-paren-newline' (remaining=180) +0ms
  eslint:rules Loading rule 'generator-star-spacing' (remaining=179) +1ms
  eslint:rules Loading rule 'implicit-arrow-linebreak' (remaining=178) +0ms
  eslint:rules Loading rule 'jsx-quotes' (remaining=177) +1ms
  eslint:rules Loading rule 'linebreak-style' (remaining=176) +0ms
  eslint:rules Loading rule 'max-statements-per-line' (remaining=175) +0ms
  eslint:rules Loading rule 'multiline-ternary' (remaining=174) +1ms
  eslint:rules Loading rule 'newline-per-chained-call' (remaining=173) +0ms
  eslint:rules Loading rule 'new-parens' (remaining=172) +1ms
  eslint:rules Loading rule 'no-floating-decimal' (remaining=171) +0ms
  eslint:rules Loading rule 'no-multi-spaces' (remaining=170) +0ms
  eslint:rules Loading rule 'no-multiple-empty-lines' (remaining=169) +1ms
  eslint:rules Loading rule 'no-trailing-spaces' (remaining=168) +0ms
  eslint:rules Loading rule 'no-whitespace-before-property' (remaining=167) +0ms
  eslint:rules Loading rule 'nonblock-statement-body-position' (remaining=166) +1ms
  eslint:rules Loading rule 'object-curly-newline' (remaining=165) +0ms
  eslint:rules Loading rule 'object-property-newline' (remaining=164) +1ms
  eslint:rules Loading rule 'one-var-declaration-per-line' (remaining=163) +0ms
  eslint:rules Loading rule 'operator-linebreak' (remaining=162) +0ms
  eslint:rules Loading rule 'padded-blocks' (remaining=161) +1ms
  eslint:rules Loading rule 'quote-props' (remaining=160) +0ms
  eslint:rules Loading rule 'rest-spread-spacing' (remaining=159) +1ms
  eslint:rules Loading rule 'semi-spacing' (remaining=158) +0ms
  eslint:rules Loading rule 'semi-style' (remaining=157) +1ms
  eslint:rules Loading rule 'space-before-function-paren' (remaining=156) +0ms
  eslint:rules Loading rule 'space-in-parens' (remaining=155) +0ms
  eslint:rules Loading rule 'space-unary-ops' (remaining=154) +1ms
  eslint:rules Loading rule 'switch-colon-spacing' (remaining=153) +0ms
  eslint:rules Loading rule 'template-curly-spacing' (remaining=152) +1ms
  eslint:rules Loading rule 'template-tag-spacing' (remaining=151) +0ms
  eslint:rules Loading rule 'wrap-iife' (remaining=150) +0ms
  eslint:rules Loading rule 'wrap-regex' (remaining=149) +1ms
  eslint:rules Loading rule 'yield-star-spacing' (remaining=148) +0ms
  eslint:rules Loading rule 'indent-legacy' (remaining=147) +1ms
  eslint:rules Loading rule 'no-spaced-func' (remaining=146) +1ms
  eslint:rules Loading rule 'arrow-body-style' (remaining=145) +0ms
  eslint:rules Loading rule 'prefer-arrow-callback' (remaining=144) +1ms
  eslint:rules Loading rule 'no-console' (remaining=143) +19ms
  eslint:rules Loading rule 'no-duplicate-imports' (remaining=142) +1ms
  eslint:rules Loading rule 'no-param-reassign' (remaining=141) +1ms
  eslint:rules Loading rule 'no-plusplus' (remaining=140) +1ms
  eslint:rules Loading rule 'no-throw-literal' (remaining=139) +1ms
  eslint:rules Loading rule 'padding-line-between-statements' (remaining=138) +0ms
  eslint:rules Loading rule 'sort-imports' (remaining=137) +2ms
  eslint:rules Loading rule 'sort-keys' (remaining=136) +1ms
  eslint:file-enumerator Yield: index.tsx +1s
  eslint:cli-engine Lint /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +0ms
  eslint:linter Linting code for /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx (pass 1) +0ms
  eslint:linter Verify +0ms
  eslint:linter With ConfigArray: /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +0ms
  eslint:linter Parsing: /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +1ms
  eslint:linter Parsing successful: /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +3s
  eslint:linter Scope analysis: /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +0ms
  eslint:linter Scope analysis successful: /Users/vladdenisov/projects/pspod/kgnc/ui/src/index.tsx +0ms
  eslint:code-path onCodePathStart s1 +0ms
  eslint:code-path onCodePathSegmentStart s1_1 +1ms
  eslint:code-path s1_1) Program +1ms
  ```
</details>



### Additional information
Also works fine when using npx or bunx without --bun (so just when using node)

**Comment by vincent1114** [comment]
Same issue, on a pipeline with latest slim version of oven. Works locally on 1.1.6.

**Comment by Costiss** [comment]
Same here 

 works on 1.1.12, but  on versions  >= 1.1.13 it hangs when using `bun run --bun` or when running on docker image 
 
```
"eslint": "^8.49.0",
"eslint-config-prettier": "^9.0.0",
"eslint-plugin-prettier": "^5.0.0",
```
```
parser: '@typescript-eslint/parser',
parserOptions: {
    ecmaVersion: 'latest',
    sourceType: 'module',
    project: './tsconfig.json',
},
plugins: ['@typescript-eslint'],
extends: [
    'prettier',
    'eslint:recommended',
    'plugin:prettier/recommended',
    'plugin:@typescript-eslint/recommended',
    'plugin:@typescript-eslint/recommended-requiring-type-checking',
    'plugin:@typescript-eslint/strict'
]
```

**Comment by joas8211** [comment]
Found the problem that is causing ESLint to hang: #13034
